File: src/main/java/com/github/wakingrufus/mastodon/ui/login/LoginController.java
package com.github.wakingrufus.mastodon.ui.login;

import com.github.wakingrufus.mastodon.events.ServerConnectEvent;
import javafx.event.Event;
import javafx.fxml.FXML;
import javafx.scene.control.Button;
import javafx.scene.control.TextField;
import lombok.extern.slf4j.Slf4j;

@Slf4j
public class LoginController {
    @FXML
    TextField serverField;

    @FXML
    Button tokenButton;

    @FXML
    public void initialize() {
        tokenButton.setOnAction(e -> {
            log.info("server: " + serverField.getText());
            Event.fireEvent(e.getTarget(), new ServerConnectEvent(e.getSource(), e.getTarget(), serverField.getText()));
        });
    }
}

File: src/main/kotlin/com/github/wakingrufus/mastodon/controllers/LoginController.kt
package com.github.wakingrufus.mastodon.controllers

import com.github.wakingrufus.mastodon.client.createServerClient
import com.github.wakingrufus.mastodon.client.registerApp
import com.sys1yagi.mastodon4j.MastodonClient
import com.sys1yagi.mastodon4j.api.entity.auth.AppRegistration
import javafx.fxml.FXML
import javafx.scene.control.Button
import javafx.scene.control.TextField
import mu.KLogging

class LoginController(private val startOAuth: (AppRegistration, MastodonClient) -> Any) : Controller<Unit> {
    companion object : KLogging()

    @FXML
    internal var serverField: TextField? = null

    @FXML
    internal var tokenButton: Button? = null

    @FXML
    override fun initialize() {
        tokenButton?.setOnAction { e ->
            logger.info("server: ${serverField?.text}")
            val client = createServerClient(serverField!!.text)
            startOAuth(registerApp(client)!!, client)
        }
    }
}